### PR TITLE
OSX CI using cmake, DMG creation support

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
This is a replacement for #11315.

Theoretically cmake CPack can be used to create the DMG. But I couldn't get that to work so instead of did it the manual hardcoded way. I'll come back to this at some point to try to switch to that. For now, this works and lets us move forward.

### CODE REVIEW

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-tags: true
+
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          create-symlink: true
+          key: ${{ runner.os }}-Debug
```
Comment (HTRamsey) [inline .github/workflows/macos.yml:41]: I forgot to set this part

key: ${{ runner.os }}-${{ matrix.BuildType }}
restore-keys: ${{ runner.os }}-${{ matrix.BuildType }}

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v3
+        with:
+          version:      6.6.3
+          aqtversion:   ==3.1.*
+          host:         mac
+          target:       desktop
+          dir:          ${{ runner.temp }}
+          modules:      qtcharts qtlocation qtpositioning qtspeech qtmultimedia qtserialport qtimageformats qtshadertools qtconnectivity qtquick3d
```
Comment (HTRamsey) [inline .github/workflows/macos.yml:54]: qt5compat

File: CMakeLists.txt [hunk 3]
```diff
@@ -295,3 +295,11 @@ if(LINUX)
     )
     install(SCRIPT "${CMAKE_SOURCE_DIR}/cmake/CreateAppImage.cmake")
 endif()
+if (MACOS)
```
Comment (HTRamsey) [inline CMakeLists.txt:298]: Not important but you could make a CreateDMG.cmake and do 
`install(SCRIPT "${CMAKE_SOURCE_DIR}/cmake/CreateDMG.cmake")`
like I did for the linux AppImage

Comment (DonLakeFlyer) [reply CMakeLists.txt:298]: Ah, right,. I'll clean that up later.

File: .github/workflows/macos.yml (hunk 2 above)
Comment (DonLakeFlyer) [reply .github/workflows/macos.yml]: Yep I added that

**Comment by HTRamsey** [comment]
I'm guessing this is replaced by #11369.

**Pull request closed without merging**